gremlin-server/src/main/java/org/apache/tinkerpop/gremlin/server/handler/HttpHandlerUtil.java

```diff
@@ -69,6 +71,20 @@ public class HttpHandlerUtil {
      * @param message The error message to contain the body.
      */
     public static void sendError(final ChannelHandlerContext ctx, final HttpResponseStatus status, final String message) {
+        sendError(ctx, status, message, false);
+    }
+
+    /**
+     * Helper method to send errors back as JSON. Only to be used when the RequestMessage couldn't be parsed, because
+     * a proper serialized ResponseMessage should be sent in that case.
+     *
+     * @param ctx           The netty channel context.
+     * @param status        The HTTP error status code.
+     * @param message       The error message to contain the body.
+     * @param closeChannel  Close the netty channel after error message.
+     */
+    public static void sendError(final ChannelHandlerContext ctx, final HttpResponseStatus status, final String message,
```

Comment:
When is closing the channel explicitly needed?

Reply:
removed.